source/postprocess/visualization/boundary_strain_rate_residual.cc

```diff
@@ -0,0 +1,127 @@
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/postprocess/visualization/boundary_strain_rate_residual.h>
+#include <aspect/postprocess/boundary_strain_rate_residual_statistics.h>
+#include <aspect/simulator.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      BoundaryStrainRateResidual<dim>::
+      BoundaryStrainRateResidual ()
+        :
+        DataPostprocessorScalar<dim> ("boundary_strain_rate_residual",
+                                      update_quadrature_points | update_gradients)
+      {}
+
+
+
+      template <int dim>
+      void
+      BoundaryStrainRateResidual<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points, ExcInternalError());
+        Assert (computed_quantities[0].size() == 1,               ExcInternalError());
+        Assert (input_data.solution_gradients[0].size() == this->introspection().n_components, ExcInternalError());
+
+        auto cell = input_data.template get_cell<dim>();
+
+        for (unsigned int q=0; q<computed_quantities.size(); ++q)
+          computed_quantities[q](0)= std::numeric_limits<double>::quiet_NaN();
+
+        const double unit_scaling_factor = this->convert_output_to_years() ? year_in_seconds : 1.0;
+
+        const Postprocess::BoundaryStrainRateResidualStatistics<dim> &boundary_strain_rate_residual_statistics =
+          this->get_postprocess_manager().template get_matching_postprocessor<Postprocess::BoundaryStrainRateResidualStatistics<dim> >();
+
+        // We only want the output at the top boundary, so only compute it if the current cell
+        // has a face at the top boundary.
+        bool cell_at_top_boundary = false;
+        for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+          if (cell->at_boundary(f) &&
+              (this->get_geometry_model().translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "top"))
+            cell_at_top_boundary = true;
+
+        if (cell_at_top_boundary)
+          for (unsigned int q=0; q<computed_quantities.size(); ++q)
+            {
+              Tensor<2,dim> grad_u;
+              for (unsigned int d=0; d<dim; ++d)
+                grad_u[d] = input_data.solution_gradients[q][d];
+              const SymmetricTensor<2,dim> strain_rate = symmetrize(grad_u);
+
+              const double data_surface_strain_rate =
+                boundary_strain_rate_residual_statistics.get_data_surface_strain_rate(input_data.evaluation_points[q]);
+
+              // Only compute residual for doubles. This condition checks for nan values
+              if (data_surface_strain_rate < 1e300)
```

Comment:
this condition is different from the statistics postprocessor, but they should be the same.